### PR TITLE
k8sfv test improvements for ongoing CI

### PR DESCRIPTION
**Pull request by neiljerram**
This PR contains the test-side changes that we need for useful ongoing CI.  By 'test-side' I mean everything that is required for the test and runs on the machine where the test is executed: the test program, Felix, API server and etcd.  The other pieces - Prometheus, push gateway and Grafana - are 'server-side' and in some senses more optional.

This PR has options to allow k8sfv to connect to those server-side pieces.  I'm now going to work on setting up those pieces for our nightly CI, and then on making it trivial for a developer to set up those pieces for themselves.

### CODE REVIEW

File: k8sfv/run-test [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# Arrange to always clean up the containers that we create.
+function cleanup {
+    # Clean up the containers that this script generates.
+    docker rm -f ${prefix}-felix || true
+    docker rm -f ${prefix}-apiserver || true
+    docker rm -f ${prefix}-etcd || true
+    # List all remaining containers.
+    docker ps -a
+}
+trap cleanup EXIT SIGINT SIGQUIT
```
Comment (fasaxc) [inline k8sfv/run-test]: Nice; I suggest making it configurable whether we run cleanup.  I've certainly found it useful to have the option to leave the rig running when I'm debugging the other FV frameworks.  WDYT?

File: k8sfv/run-test [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+# Wait until the newly launched API server can respond to a request,
+# before continuing.
+while ! curl -k ${k8s_api_endpoint}/apis/extensions/v1beta1/thirdpartyresources; do
+    sleep 2
+done
+
+# Run Felix.
+mkdir -p k8sfv/output
+docker run --detach --privileged --name ${prefix}-felix \
+       -e FELIX_LOGSEVERITYSCREEN=info \
```
Comment (fasaxc) [inline k8sfv/run-test]: Worth making the log level configurable?

File: k8sfv/run-test (hunk 1 above)
Comment (neiljerram) [inline k8sfv/run-test]: I've coincidentally just added sthg for this in parallel - please see SLEEP_BEFORE_CLEANUP in the commits I'm just about to add.

**Comment by neiljerram** [comment]
@fasaxc Would you like to review the deltas since your previous review above?

FYI, also:
- we've now had a clean-looking CI run, with this latest code, at http://localhost:8080/job/k8sfv/11/console (and in which even the leak test passed!)
- I'm working as you know on the ongoing Grafana/Prometheus/push gateway setup.

**Pull request merged**